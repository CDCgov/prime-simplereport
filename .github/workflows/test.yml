name: Test

# Run this on pushes to any branch that change a backend file or the workflow definition
on:
  workflow_dispatch: # because sometimes you just want to force a branch to have tests run
  pull_request:
    branches:
      - "**"
  push:
    branches:
      - main

env:
  NODE_VERSION: 14
  JAVA_VERSION: 11
  PROJECT_ROOT: /home/runner/work/prime-simplereport/prime-simplereport # Hardcoding this here because env.WORKSPACE_ROOT gets overridden by one of the steps downstream. We only need this for Sonar.

defaults:
  run:
    working-directory: backend

jobs:
  test:
    runs-on: ubuntu-latest
    services:
      test-db:
        image: postgres:11-alpine
        env:
          POSTGRES_PASSWORD: this_is_a_super_secure_admin_password
        # Set health checks to wait until postgres has started
        options: >-
          --health-cmd pg_isready
          --health-interval 10s
          --health-timeout 5s
          --health-retries 5
        ports:
          - 5432:5432
    steps:
      - uses: actions/checkout@v2
        with:
          fetch-depth: 0 # Disable shallow clones so Sonar can have all the data
      - name: Set up JDK ${{env.JAVA_VERSION}}
        uses: actions/setup-java@v1
        with:
          java-version: ${{env.JAVA_VERSION}}
      - name: Cache Java Dependencies
        uses: actions/cache@v2
        with:
          path: |
            ~/.gradle/caches
            ~/.gradle/wrapper
          key: gradle-${{ hashFiles('*.gradle', 'gradle/dependency-locks/*') }}
      - name: Set up postgresql sidecar
        env:
          PGHOST: localhost
          PGPASSFILE: db-setup/.pgpass_ci
          POSTGRES_USER: postgres
          POSTGRES_DB: postgres
          LIB_DIR: db-setup
        run: |
          chmod 0600 $PGPASSFILE
          db-setup/create-db.sh
      - name: Run Java Tests
        env:
          OKTA_TESTING_DISABLEHTTPS: true
          TWILIO_ACCOUNT_SID: ${{secrets.TWILIO_TEST_ACCOUNT_SID }}
          TWILIO_AUTH_TOKEN: ${{secrets.TWILIO_TEST_AUTH_TOKEN }}
        run: ./gradlew jacocoTestReport -PtestDbPort=5432
      - name: Archive Test Results
        uses: actions/upload-artifact@v2
        if: failure()
        with:
          name: test-report
          path: backend/build/reports/tests/test
          retention-days: 7
      - name: Use Node.js ${{env.NODE_VERSION}}
        uses: actions/setup-node@v2.1.5
        with:
          node-version: ${{env.NODE_VERSION}}
      - name: Cache yarn
        uses: actions/cache@v2
        with:
          path: ~/.cache/yarn
          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            ${{ runner.os }}-yarn-
      - name: Node setup
        working-directory: ./frontend
        run: yarn install --prefer-offline
      - name: Test
        working-directory: ./frontend
        run: yarn test:ci
      - name: "Function Apps: npm install"
        working-directory: ./ops/services/app_functions/report_stream_batched_publisher/functions
        run: npm ci
      - name: "Function Apps: Test"
        working-directory: ./ops/services/app_functions/report_stream_batched_publisher/functions
        run: npm run coverage
      - name: Sonar analysis
        if: ${{ github.actor != 'dependabot[bot]' }}
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # Needed to get PR information, if any
          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
        run: ./gradlew sonarqube -Dsonar.projectBaseDir=${{ env.PROJECT_ROOT }} --info
  build-jar:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Set up JDK ${{env.JAVA_VERSION}}
        uses: actions/setup-java@v1
        with:
          java-version: ${{env.JAVA_VERSION}}
      - name: Cache Java Dependencies
        uses: actions/cache@v2
        with:
          path: |
            ~/.gradle/caches
            ~/.gradle/wrapper
          key: gradle-${{ hashFiles('*.gradle', 'gradle/dependency-locks/*') }}
      - name: Run Java Build
        run: ./gradlew assemble
  backend-lint:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Set up JDK ${{env.JAVA_VERSION}}
        uses: actions/setup-java@v1
        with:
          java-version: ${{env.JAVA_VERSION}}
      - name: Cache Java Dependencies
        uses: actions/cache@v2
        with:
          path: |
            ~/.gradle/caches
            ~/.gradle/wrapper
          key: gradle-${{ hashFiles('*.gradle', 'gradle/dependency-locks/*') }}
      - name: Run Style checks
        run: ./gradlew checkstyleMain checkstyleTest spotlessCheck --continue
  frontend-lint-and-build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Use Node.js ${{env.NODE_VERSION}}
        uses: actions/setup-node@v2.1.5
        with:
          node-version: ${{env.NODE_VERSION}}
      - name: Cache yarn
        uses: actions/cache@v2
        with:
          path: ~/.cache/yarn
          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            ${{ runner.os }}-yarn-
      - name: Node setup
        working-directory: ./frontend
        run: yarn
      - name: Lint front-end
        working-directory: ./frontend
        run: yarn lint
      - name: Build release
        working-directory: ./frontend
        run: yarn build
