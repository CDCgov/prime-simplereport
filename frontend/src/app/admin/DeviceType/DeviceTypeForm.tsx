import React, { useState } from "react";

import Button from "../../commonComponents/Button/Button";
import TextInput from "../../commonComponents/TextInput";

import { Device } from "./DeviceTypeFormContainer";

interface Props {
  saveDeviceType: (device: Device) => void;
}

const DeviceTypeForm: React.FC<Props> = ({ saveDeviceType }) => {
  const [device, updateDevice] = useState<Device>({
    name: "",
    manufacturer: "",
    model: "",
    loincCode: "",
    swabType: "",
  });
  const [formChanged, updateFormChanged] = useState<boolean>(false);
  const onChange = (
    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
  ) => {
    updateDevice({ ...device, [e.target.name]: e.target.value });
    updateFormChanged(true);
  };
  return (
    <main className="prime-home">
      <div className="grid-container">
        <div className="grid-row">
          <div className="prime-container card-container">
            <div className="usa-card__header">
              <div>
                <h2>Device Type</h2>
              </div>
              <div
                style={{
                  display: "flex",
                  justifyContent: "center",
                  alignItems: "center",
                }}
              >
                <Button
                  type="button"
                  onClick={() => saveDeviceType(device)}
                  label="Save Changes"
                  disabled={!formChanged}
                />
              </div>
            </div>
            <div className="usa-card__body margin-top-1">
              <div>
                <div className="usa-alert usa-alert--warning">
                  <div className="usa-alert__body">
                    <h4 className="usa-alert__heading">Reminder</h4>
                    <p className="usa-alert__text">
                      Notify ReportStream of devices added in{" "}
                      <a
                        href="https://usds.slack.com/archives/C024MGSJZ38"
                        target="_blank"
                      >
                        #prime-reportstream
                      </a>
                    </p>
                    <p>
                      Device details can be found by downloading the excel file
                      published at{" "}
                      <a
                        href="https://www.cdc.gov/csels/dls/sars-cov-2-livd-codes.html"
                        target="_blank"
                      >
                        https://www.cdc.gov/csels/dls/sars-cov-2-livd-codes.html
                      </a>
                    </p>
                  </div>
                </div>
              </div>
              <div className="grid-row grid-gap">
                <div className="tablet:grid-col">
                  <TextInput
                    label="Name"
                    name="name"
                    value={device.name}
                    onChange={onChange}
                    required
                  />
                </div>
                <div className="tablet:grid-col">
                  <TextInput
                    label="Manufacturer"
                    name="manufacturer"
                    value={device.manufacturer}
                    onChange={onChange}
                    required
                  />
                </div>
                <div className="tablet:grid-col">
                  <TextInput
                    label="Model"
                    name="model"
                    value={device.model}
                    onChange={onChange}
                    required
                  />
                </div>
                <div className="tablet:grid-col">
                  <TextInput
                    label="Loinc Code"
                    name="loincCode"
                    value={device.loincCode}
                    onChange={onChange}
                    required
                  />
                </div>
              </div>
              <div className="grid-row grid-gap">
                <div className="tablet:grid-col">
                  <TextInput
                    label="SNOMED code of Swab Type"
                    name="swabType"
                    value={device.swabType}
                    onChange={onChange}
                    required
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  );
};

export default DeviceTypeForm;
