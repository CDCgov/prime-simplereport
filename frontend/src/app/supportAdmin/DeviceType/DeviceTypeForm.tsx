import React, { useState } from "react";

import Button from "../../commonComponents/Button/Button";
import TextInput from "../../commonComponents/TextInput";
import MultiSelect from "../../commonComponents/MultiSelect/MultiSelect";
import { MultiSelectDropdownOption } from "../../commonComponents/MultiSelect/MultiSelectDropdown/MultiSelectDropdown";

import { Device } from "./DeviceTypeFormContainer";
import DeviceTypeReminderMessage from "./DeviceTypeReminderMessage";

interface Props {
  saveDeviceType: (device: Device) => void;
  swabOptions: Array<MultiSelectDropdownOption>;
  supportedDiseaseOptions: Array<MultiSelectDropdownOption>;
}

const DeviceTypeForm: React.FC<Props> = ({
  saveDeviceType,
  swabOptions,
  supportedDiseaseOptions,
}) => {
  const [device, updateDevice] = useState<Device>({
    name: "",
    manufacturer: "",
    model: "",
    loincCode: "",
    swabTypes: [],
    supportedDiseases: [],
  });
  const [formChanged, updateFormChanged] = useState<boolean>(false);

  function updateDeviceAttribute(name: string, value: any) {
    updateDevice({ ...device, [name]: value });
    updateFormChanged(true);
  }

  const onChange = (
    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
  ) => {
    updateDeviceAttribute(e.target.name, e.target.value);
  };
  return (
    <main className="prime-home">
      <div className="grid-container">
        <div className="grid-row">
          <div className="prime-container card-container">
            <div className="usa-card__header">
              <div>
                <h2>Device type</h2>
              </div>
              <div
                style={{
                  display: "flex",
                  justifyContent: "center",
                  alignItems: "center",
                }}
              >
                <Button
                  type="button"
                  onClick={() => saveDeviceType(device)}
                  label="Save changes"
                  disabled={!formChanged}
                />
              </div>
            </div>
            <div className="usa-card__body margin-top-1">
              <DeviceTypeReminderMessage />
              <div className="grid-row grid-gap">
                <div className="tablet:grid-col">
                  <TextInput
                    label="Device name"
                    name="name"
                    value={device.name}
                    onChange={onChange}
                    required
                  />
                </div>
                <div className="tablet:grid-col">
                  <TextInput
                    label="Manufacturer"
                    name="manufacturer"
                    value={device.manufacturer}
                    onChange={onChange}
                    required
                  />
                </div>
                <div className="tablet:grid-col">
                  <TextInput
                    label="Model"
                    name="model"
                    value={device.model}
                    onChange={onChange}
                    required
                  />
                </div>
                <div className="tablet:grid-col">
                  <TextInput
                    label="LOINC code"
                    name="loincCode"
                    value={device.loincCode}
                    onChange={onChange}
                    required
                  />
                </div>
              </div>
              <div className="grid-row grid-gap">
                <div className="tablet:grid-col">
                  <MultiSelect
                    label="SNOMED code for swab type(s)"
                    name="swabTypes"
                    onChange={(swabTypes) => {
                      updateDeviceAttribute("swabTypes", swabTypes);
                    }}
                    options={swabOptions}
                    required
                  />
                </div>
              </div>
              <div className="grid-row grid-gap">
                <div
                  className="tablet:grid-col"
                  style={{ marginBottom: "56px" }}
                >
                  <MultiSelect
                    label="Supported diseases"
                    name="supportedDiseases"
                    onChange={(supportedDiseases) => {
                      updateDeviceAttribute(
                        "supportedDiseases",
                        supportedDiseases
                      );
                    }}
                    options={supportedDiseaseOptions}
                    required
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  );
};

export default DeviceTypeForm;
