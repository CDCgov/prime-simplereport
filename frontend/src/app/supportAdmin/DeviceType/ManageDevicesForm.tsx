import React, { useState } from "react";

import Button from "../../commonComponents/Button/Button";
import TextInput from "../../commonComponents/TextInput";
import MultiSelect from "../../commonComponents/MultiSelect/MultiSelect";
import Select from "../../commonComponents/Select";
import { MultiSelectDropdownOption } from "../../commonComponents/MultiSelect/MultiSelectDropdown/MultiSelectDropdown";
import { DeviceType, UpdateDeviceType } from "../../../generated/graphql";

import DeviceTypeReminderMessage from "./DeviceTypeReminderMessage";

interface Props {
  updateDeviceType: (device: UpdateDeviceType) => void;
  swabOptions: Array<MultiSelectDropdownOption>;
  supportedDiseaseOptions: Array<MultiSelectDropdownOption>;
  devices: DeviceType[];
}

const ManageDevicesForm: React.FC<Props> = ({
  updateDeviceType,
  swabOptions,
  supportedDiseaseOptions,
  devices,
}) => {
  const [selectedDevice, setSelectedDevice] = useState<
    UpdateDeviceType | undefined
  >(undefined);

  const [formChanged, updateFormChanged] = useState<boolean>(false);

  const updateDeviceAttribute = (name: string, value: any) => {
    if (selectedDevice) {
      setSelectedDevice({ ...selectedDevice, [name]: value });
    }
  };

  const onChange = (
    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
  ) => {
    updateDeviceAttribute(e.target.name, e.target.value);
  };

  const getDeviceNames = () =>
    Array.from(
      devices.map((device) => ({
        label: device.name,
        value: device.internalId,
      }))
    );

  const getUpdateDeviceType = (
    device?: DeviceType
  ): UpdateDeviceType | undefined => {
    return device
      ? {
          internalId: device.internalId,
          name: device.name,
          manufacturer: device.manufacturer,
          model: device.model,
          swabTypes: device.swabTypes?.map((swab) => swab.internalId),
          supportedDiseases:
            device.supportedDiseases?.map((disease) => disease.internalId) ||
            [],
          loincCode: device.loincCode,
        }
      : undefined;
  };

  return (
    <main className="prime-home">
      <div className="grid-container">
        <div className="grid-row">
          <div className="prime-container card-container">
            <div className="usa-card__header">
              <div>
                <h2>Manage devices</h2>
              </div>
              <div
                style={{
                  display: "flex",
                  justifyContent: "center",
                  alignItems: "center",
                }}
              >
                <Button
                  type="button"
                  onClick={() =>
                    selectedDevice && updateDeviceType(selectedDevice)
                  }
                  label="Save changes"
                  disabled={!formChanged || !selectedDevice}
                />
              </div>
            </div>
            <div className="usa-card__body margin-top-1">
              <DeviceTypeReminderMessage />
              <div className="grid-row grid-gap">
                <div className="tablet:grid-col">
                  <Select
                    label="Device name"
                    name="name"
                    value={selectedDevice?.internalId || ""}
                    options={getDeviceNames()}
                    defaultSelect
                    onChange={(id) => {
                      updateFormChanged(!!id);
                      setSelectedDevice(
                        getUpdateDeviceType(
                          devices.find((device) => id === device.internalId)
                        )
                      );
                    }}
                    required
                  />
                </div>
                <div className="tablet:grid-col">
                  <TextInput
                    label="Manufacturer"
                    name="manufacturer"
                    value={selectedDevice?.manufacturer}
                    onChange={onChange}
                    disabled={!selectedDevice}
                    required
                  />
                </div>
                <div className="tablet:grid-col">
                  <TextInput
                    label="LOINC code"
                    name="loincCode"
                    value={selectedDevice?.loincCode}
                    onChange={onChange}
                    disabled={!selectedDevice}
                    required
                  />
                </div>
              </div>
              <div className="grid-row grid-gap">
                <div className="tablet:grid-col">
                  <TextInput
                    label="Model"
                    name="model"
                    value={selectedDevice?.model}
                    onChange={onChange}
                    disabled={!selectedDevice}
                    required
                  />
                </div>
              </div>
              <div className="grid-row grid-gap">
                <div className="tablet:grid-col">
                  <MultiSelect
                    key={selectedDevice?.internalId}
                    label="SNOMED code for swab type(s)"
                    name="swabTypes"
                    onChange={(swabTypes) => {
                      updateDeviceAttribute("swabTypes", swabTypes);
                    }}
                    options={swabOptions}
                    initialSelectedValues={selectedDevice?.swabTypes}
                    disabled={!selectedDevice}
                    required
                  />
                </div>
              </div>
              <div className="grid-row grid-gap">
                <div
                  className="tablet:grid-col"
                  style={{ marginBottom: "56px" }}
                >
                  <MultiSelect
                    key={selectedDevice?.internalId}
                    label="Supported diseases"
                    name="supportedDiseases"
                    onChange={(supportedDiseases) => {
                      updateDeviceAttribute(
                        "supportedDiseases",
                        supportedDiseases
                      );
                    }}
                    options={supportedDiseaseOptions}
                    initialSelectedValues={selectedDevice?.supportedDiseases}
                    disabled={!selectedDevice}
                    required
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  );
};

export default ManageDevicesForm;
