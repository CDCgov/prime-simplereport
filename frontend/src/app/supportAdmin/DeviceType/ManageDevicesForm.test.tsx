import { render, screen, within } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import ManageDevicesForm from "./ManageDevicesForm";
import { addValue } from "./DeviceTypeForm.test";

describe("ManageDeviceTypeForm", () => {
  let saveDeviceType: jest.Mock;

  beforeEach(() => {
    saveDeviceType = jest.fn();
    render(
      <ManageDevicesForm
        updateDeviceType={saveDeviceType}
        swabOptions={[
          { label: "nose", value: "123" },
          { label: "eye", value: "456" },
          { label: "mouth", value: "789" },
        ]}
        supportedDiseaseOptions={[{ label: "COVID-19", value: "123" }]}
        devices={[
          {
            internalId: "abc1",
            name: "Tesla Emitter",
            model: "Model A",
            manufacturer: "Celoxitin",
            loincCode: "1234-1",
            swabTypes: [{ internalId: "123", name: "nose", typeCode: "n123" }],
            supportedDiseases: [
              { internalId: "123", name: "COVID-19", loinc: "1234-1" },
            ],
          },
          {
            internalId: "abc2",
            name: "Fission Energizer",
            model: "Model B",
            manufacturer: "Curentz",
            loincCode: "1234-2",
            swabTypes: [{ internalId: "456", name: "eye", typeCode: "e456" }],
            supportedDiseases: [
              { internalId: "123", name: "COVID-19", loinc: "1234-1" },
            ],
          },
          {
            internalId: "abc3",
            name: "Covalent Observer",
            model: "Model C",
            manufacturer: "Vitamin Tox",
            loincCode: "1234-3",
            swabTypes: [{ internalId: "789", name: "mouth", typeCode: "m789" }],
            supportedDiseases: [
              { internalId: "123", name: "COVID-19", loinc: "1234-1" },
            ],
          },
        ]}
      />
    );
  });

  it("Disables the save button", () => {
    expect(screen.getByText("Save changes")).toBeDisabled();
  });

  it("disables all input fields when no device is selected", function () {
    expect(
      screen.getByLabelText("Manufacturer", { exact: false })
    ).toBeDisabled();
    expect(screen.getByLabelText("Model", { exact: false })).toBeDisabled();
    expect(
      screen.getByLabelText("LOINC code", { exact: false })
    ).toBeDisabled();
    expect(screen.getAllByTestId("multi-select-toggle")[0]).toBeDisabled();
  });

  it("shows a list of devices to select from", () => {
    userEvent.click(screen.getByLabelText("Device name", { exact: false }));
    expect(screen.getByText("Tesla Emitter")).toBeInTheDocument();
    expect(screen.getByText("Fission Energizer")).toBeInTheDocument();
    expect(screen.getByText("Covalent Observer")).toBeInTheDocument();
  });

  describe("When selecting a device", () => {
    beforeEach(() => {
      userEvent.selectOptions(
        screen.getByLabelText("Device name", { exact: false }),
        "Tesla Emitter"
      );
    });

    it("enables input fields and prefills them with current values", () => {
      const manufacturerInput = screen.getByLabelText("Manufacturer", {
        exact: false,
      });
      const modelInput = screen.getByLabelText("Model", { exact: false });
      const loincCodeInput = screen.getByLabelText("LOINC code", {
        exact: false,
      });
      const snomedInput = screen.getAllByTestId("multi-select-toggle")[0];
      const pillContainer = screen.getAllByTestId("pill-container")[0];

      expect(manufacturerInput).toBeEnabled();
      expect(modelInput).toBeEnabled();
      expect(loincCodeInput).toBeEnabled();
      expect(snomedInput).toBeEnabled();

      expect(manufacturerInput).toHaveValue("Celoxitin");
      expect(modelInput).toHaveValue("Model A");
      expect(loincCodeInput).toHaveValue("1234-1");
      within(pillContainer).getByText("nose");
    });

    it("displays a list of available snomeds", () => {
      const snomedList = screen.getAllByTestId("multi-select-option-list")[0];

      expect(within(snomedList).getByText("eye")).toBeInTheDocument();
      expect(within(snomedList).getByText("mouth")).toBeInTheDocument();
      expect(within(snomedList).getByText("nose")).toBeInTheDocument();
    });

    describe("selecting another device", () => {
      beforeEach(() => {
        userEvent.selectOptions(
          screen.getByLabelText("Device name", { exact: false }),
          "Fission Energizer"
        );
      });

      it("prefills input fields with new values", () => {
        const manufacturerInput = screen.getByLabelText("Manufacturer", {
          exact: false,
        });
        const modelInput = screen.getByLabelText("Model", { exact: false });
        const loincCodeInput = screen.getByLabelText("LOINC code", {
          exact: false,
        });
        const pillContainer = screen.getAllByTestId("pill-container", {
          exact: false,
        })[0];

        expect(manufacturerInput).toHaveValue("Curentz");
        expect(modelInput).toHaveValue("Model B");
        expect(loincCodeInput).toHaveValue("1234-2");
        within(pillContainer).getByText("eye");
      });
    });

    describe("updating a device", () => {
      it("calls update device with the current values", () => {
        const snomedInput = screen.getAllByTestId("multi-select-toggle")[0];
        const snomedList = screen.getAllByTestId("multi-select-option-list")[0];

        addValue("Manufacturer", " LLC");
        addValue("Model", "X");
        addValue("LOINC code", "234");
        userEvent.click(snomedInput);
        userEvent.click(within(snomedList).getByText("eye"));
        userEvent.click(screen.getByText("Save changes"));

        expect(saveDeviceType).toHaveBeenNthCalledWith(1, {
          internalId: "abc1",
          name: "Tesla Emitter",
          model: "Model AX",
          manufacturer: "Celoxitin LLC",
          loincCode: "1234-1234",
          swabTypes: ["123", "456"],
          supportedDiseases: ["123"],
        });
        expect(saveDeviceType).toBeCalledTimes(1);
      });
    });
  });
});
