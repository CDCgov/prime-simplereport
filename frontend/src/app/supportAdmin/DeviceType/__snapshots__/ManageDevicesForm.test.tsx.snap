// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageDeviceTypeForm renders the Manage Device Type Form Container item 1`] = `undefined`;
